src/library/tags/facetid.cpp

```diff
@@ -0,0 +1,176 @@
+#include "library/tags/facetid.h"
+
+#include <QRegularExpression>
+
+namespace {
+
+const QRegularExpression kValidFacetStringNotEmpty(
+        QStringLiteral("^[\\+\\-\\./0-9@a-z\\[\\]_]+"));
```

Comment:
Should we enforce starting with lowercase alphabetic letter? I think facet IDs like `_` would be weird. Just my 2 cents. 

```suggestion
        QStringLiteral("^[a-z][\\+\\-\\./0-9@a-z\\[\\]_]*"));
```